dsws_client/parse.py

```diff
@@ -63,19 +63,25 @@ def responses_to_records(
     responses: Iterable[DSDataResponse],
     *,
     process_strings: bool = True,
-) -> ParsedResponse:
+) -> Iterator[ParsedResponse]:
     """Parse a list of DSDataResponse objects into a list of records."""
-    parsed_response = ParsedResponse()
     for response in responses:
         try:
```

Comment:
if in this func enters the actual response, here you would get the `data_response` from it and then do whatever you need after?

Reply:
The actual response is not one response, but several responses, each with several data responses inside. So it's a list of lists, that's why I use `chain.from_iterable` to flatten them. But `responses_to_records` should only be responsible for parsing data responses, that's why we're flattening the responses before passing them.

Reply:
I see ok